packages/clerk-js/src/ui/hooks/useDebounce.ts

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useState } from 'react';
+
+export function useDebounce<T>(value: T, delay?: number): T {
```

Comment:
🔧 Consider renaming this hook to useSetTimeout as it's not strictly about debouncing user input.